File: 백준/Gold/3649. 로봇 프로젝트/로봇 프로젝트.py
import sys

while True:
    try:
        x = int(sys.stdin.readline().rstrip())
        n = int(sys.stdin.readline().rstrip())
        l = []

        for i in range(n):
            l.append(int(sys.stdin.readline().rstrip()))

        l.sort()
        x *= 10e6
        min_pointer = 0
        max_pointer = n-1

        while min_pointer < max_pointer:
            if l[min_pointer] + l[max_pointer] == x:
                print("yes", l[min_pointer], l[max_pointer])
                break
            elif l[min_pointer] + l[max_pointer] < x:
                min_pointer += 1
            else:
                max_pointer -= 1
        else:
            print("danger")
    
    except:
        break

File: 백준/Gold/3649. 로봇 프로젝트/로봇 프로젝트.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String input;

        while ((input = br.readLine()) != null) {
            try {
                boolean found = false;
                int X = Integer.parseInt(input);
                X *= 10000000;

                input = br.readLine();
                if (input == null) break;
                int n = Integer.parseInt(input);

                List<Integer> blocks = new ArrayList<>();

                for (int i = 0; i < n; i++) {
                    input = br.readLine();
                    if (input == null) break;
                    blocks.add(Integer.parseInt(input));
                }

                Collections.sort(blocks);

                int left = 0;
                int right = blocks.size() - 1;

                while (left < right) {
                    int l = blocks.get(left);
                    int r = blocks.get(right);
                    if (l + r == X) {
                        System.out.println("yes " + l + " " + r);
                        found = true;
                        break;
                    } else if (l + r < X) {
                        left++;
                    } else {
                        right--;
                    }
                }

                if (!found) {
                    System.out.println("danger");
                }
            } catch (NumberFormatException e) {
                return;
            }
        }
    }
}